cibuildwheel/linux.py

```diff
@@ -138,6 +138,13 @@ def build(options: BuildOptions) -> None:
                     log.build_start(config.identifier)
 
                     dependency_constraint_flags: List[Union[str, PathLike]] = []
+                    if config.identifier.startswith("pp"):
+                        # Patch PyPy to make sure headers get installed into a venv
+                        patch_version = '_27' if config.version == '2.7' else ''
+                        patch_path = Path(__file__).absolute().parent / 'resources' / f'pypy_venv{patch_version}.patch'
+                        patch_docker_path = PurePath('/pypy_venv.patch')
+                        docker.copy_into(patch_path, patch_docker_path)
+                        docker.call(['patch', '-N', '-d', config.path, patch_docker_path])
```

Comment:
This is much neater :)

Reply:
It's only slightly duplicated, now, in `macos.py`, since I removed the `pypy_patch` function. But I think I prefer this clarify.